machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +166,161 @@ type NutanixResourceIdentifier struct {
 	Name *string `json:"name,omitempty"`
 }
 
+// NutanixGPUIdentifierType is an enumeration of different resource identifier types for GPU entities.
+type NutanixGPUIdentifierType string
+
+const (
+	// NutanixGPUIdentifierName identifies a GPU by Name.
+	NutanixGPUIdentifierName NutanixGPUIdentifierType = "Name"
+
+	// NutanixGPUIdentifierDeviceID identifies a GPU by device ID.
+	NutanixGPUIdentifierDeviceID NutanixGPUIdentifierType = "DeviceID"
+)
+
+// NutanixGPU holds the identity of a Nutanix GPU resource in the Prism Central
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'DeviceID' ?  has(self.deviceID) : !has(self.deviceID)",message="deviceID configuration is required when type is DeviceID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Name' ?  has(self.name) : !has(self.name)",message="name configuration is required when type is Name, and forbidden otherwise"
+// +union
+type NutanixGPU struct {
+	// type is the identifier type to use for this GPU resource.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NutanixGPUIdentifierType `json:"type"`
+
+	// deviceID is the id of the GPU entity.
+	// +optional
+	// +unionMember
+	DeviceID *int64 `json:"deviceID,omitempty"`
+
+	// name is the GPU name
```

Comment:
What is the format here, max length, min length, valid characters etc?

Reply:
I cannot find any restrictions for the GPU device name in the prism-api documentation. Since the "name" here is used to find the existing GPU device but not to create a new one,  I think it should be fine not to put the naming restrictions here.

Reply:
So i could create a GPU with the name !@£$@!%$£@? 😅  Might be worth double checking